BasicSample/Assets/ARAnchor/Scripts/SampleAnchor.cs

```diff
@@ -20,45 +21,40 @@ public class SampleAnchor : MonoBehaviour
         [SerializeField]
         private Material transientAnchorMaterial = null;
 
-        private ARAnchor m_arAnchor;
-        private string m_name = "";
-        public string Name
-        {
-            get => m_name;
-            set
-            {
-                m_name = value;
-                UpdateText();
-            }
-        }
+        public string Name { get; set; }
 
+        private ARAnchor m_arAnchor;
         private bool m_persisted = false;
         public bool Persisted
         {
             get => m_persisted;
             set
             {
-                m_persisted = value;
-                meshRenderer.material = m_persisted ? persistentAnchorMaterial : transientAnchorMaterial;
-                UpdateText();
+                if (m_persisted != value)
+                {
+                    m_persisted = value;
+                    meshRenderer.material = m_persisted ? persistentAnchorMaterial : transientAnchorMaterial;
+                }
             }
         }
 
-        private void UpdateText()
+        private void Start()
         {
-            if (m_arAnchor == null) return;
-            text.text = (Persisted ? $"\"{m_name}\": " : "") + m_arAnchor.trackableId.ToString();
+            m_arAnchor = GetComponent<ARAnchor>();
         }
 
-        private void Start()
+        private void Update()
         {
-            m_arAnchor = GetComponent<ARAnchor>();
-            if (m_arAnchor == null)
+            string info = Persisted ? $"\"{Name}\": " : "";
+            if (m_arAnchor != null)
+            {
+                info = $"{m_arAnchor.trackableId}\n{m_arAnchor.trackingState} " + info;
```

Comment:
Allocating a string in the update loop, especially one that might not be used, feels unnecessarily heavy on the GC. I have a slightly different approach in #70, where I keep track of the anchors reporting themselves as updated before changing the display string. I can clean up that PR later today if we need this feature ASAP.

Reply:
Got it.  I will modify to your pattern.  I will only take the part that I need here.  Will count on yours for the visual updates.

BTW: no need to break into working mode during vacation, I will do my best to get your satisfied.

Reply:
~~Stole~~ Borrowed the idea from @keveleigh 's change to void text update when there's nothing changed.  I changed it a bit to use dirty flag instead of inline update, to avoid minor unnecessary when properties are changed in batch.